### ISSUE TITLE
Support dependencies (`require`) between tests

### ISSUE THREAD

**Issue by comps** [teemtee/tmt#1897]
Suppose there are two tests called

* `/check-xml-syntax`
* `/check-standards-compliance`

that both consume data from some long-running operation, ie. a VM installation and a full-OS-scan by some tool:

* `/scan-full-os-in-VM`

To avoid that lengthy operation running twice, the `/scan-full-os-in-VM` test exports all scanning details in some deterministic location, where the other 2 checks can find it, ie. `/var/tmp/scan-results`.

However this means some logic (TMT) needs to ensure that, whenever the user specifies either of the two checks to be run, the `/scan-full-os-in-VM` test must be executed prior to the checks running. For example by extending `require`:

```
require:
  - test: /scan-full-os-in-VM
```

(I imagine it would be possible to `discover` and run tests from other repositories too in this way, without any extra work.)

Like with RPM `require`s, the required test should be added only once, before any and all tests that `require` it.

**Comment by psss** [comment]
Sorry for the late response. @comps, are you still interested in the feature? Sounds like and interesting idea. Could be useful. Might be also related to #825. Let's discuss this on the hacking session.

**Comment by comps** [comment]
I don't think we need this anymore; I believe most tmt users have gotten used to splitting workflow based on plans, ie. to have plans per "test type". If some tests require virtualization, to group those tests to a common plan that requires HVM-capable hardware in `provision`.  
I'm not sure if that's the best approach, but (from asking around), that's what we naturally seemed to have done.

Similarly, the `prepare` section of a plan can be used as a setup (and `finish` as cleanup) for code related to that use case (ie. setting up a libvirt host).

And Beakerlib libraries / other test library or module logic can cover code reuse. Ie. having some setup inside `common.sh` and sourcing it from each test in a given directory.

**Comment by psss** [comment]
Ok, thank you, closing, we can always revisit if needed.
